Sources/_StringProcessing/RegexDSL/Assertion.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import _MatchingEngine
+
+public struct Anchor {
+  internal enum Kind {
+    case startOfSubject
+    case endOfSubjectBeforeNewline
+    case endOfSubject
+    case firstMatchingPositionInSubject
+    case textSegmentBoundary
+    case startOfLine
+    case endOfLine
+    case wordBoundary
+  }
+  
+  var kind: Kind
+  var isInverted: Bool = false
+}
+
+extension Anchor: RegexProtocol {
+  var astAssertion: AST.Atom.AssertionKind {
+    if !isInverted {
+      switch kind {
+      case .startOfSubject: return .startOfSubject
+      case .endOfSubjectBeforeNewline: return .endOfSubjectBeforeNewline
+      case .endOfSubject: return .endOfSubject
+      case .firstMatchingPositionInSubject: return .firstMatchingPositionInSubject
+      case .textSegmentBoundary: return .textSegment
+      case .startOfLine: return .startOfLine
+      case .endOfLine: return .endOfLine
+      case .wordBoundary: return .wordBoundary
+      }
+    } else {
+      switch kind {
+      case .startOfSubject: fatalError("Not yet supported")
+      case .endOfSubjectBeforeNewline: fatalError("Not yet supported")
+      case .endOfSubject: fatalError("Not yet supported")
+      case .firstMatchingPositionInSubject: fatalError("Not yet supported")
+      case .textSegmentBoundary: return .notTextSegment
+      case .startOfLine: fatalError("Not yet supported")
+      case .endOfLine: fatalError("Not yet supported")
+      case .wordBoundary: return .notWordBoundary
+      }
+    }
+  }
+  
+  public var regex: Regex<Substring> {
+    Regex(node: .atom(.assertion(astAssertion)))
+  }
+}
+
+// MARK: - Public API
+
+extension Anchor {
+  public static var startOfSubject: Anchor {
+    Anchor(kind: .startOfSubject)
+  }
+
+  public static var endOfSubjectBeforeNewline: Anchor {
+    Anchor(kind: .endOfSubjectBeforeNewline)
+  }
+
+  public static var endOfSubject: Anchor {
+    Anchor(kind: .endOfSubject)
+  }
+
+  // TODO: Are we supporting this?
+//  public static var resetStartOfMatch: Anchor {
+//    Anchor(kind: resetStartOfMatch)
+//  }
+
+  public static var firstMatchingPositionInSubject: Anchor {
+    Anchor(kind: .firstMatchingPositionInSubject)
+  }
+
+  public static var textSegmentBoundary: Anchor {
+    Anchor(kind: .textSegmentBoundary)
+  }
+  
+  public static var startOfLine: Anchor {
+    Anchor(kind: .startOfLine)
+  }
+
+  public static var endOfLine: Anchor {
+    Anchor(kind: .endOfLine)
+  }
+
+  public static var wordBoundary: Anchor {
+    Anchor(kind: .wordBoundary)
+  }
+  
+  public var inverted: Anchor {
+    var result = self
+    result.isInverted.toggle()
+    return result
+  }
+}
+
+public func lookahead<R: RegexProtocol>(
+  isNegative: Bool = false,
```

Comment:
While generally Boolean properties should read like an assertion about the receiver, I'm not sure it's quite as useful for function parameters. In this case, `isNegative` is not forming a phrase with the base name to produce Boolean result, but a dictation by the caller to modify the behavior of the callee. As such, I wonder if we should call it `negative` instead.

Reply:
You're right, this is like the second parameter in `split(separator: "-", omittingEmptySubsequences: false)`. 👍🏻